doc/developer/design/20230210_storage_persist_sink_redux.md

```diff
@@ -0,0 +1,197 @@
+appended to a shard. We only emit finalized batches to the `append_batches` operator when that batch is contained within
+a batch descriptions that we learned about from `mint_batch_descriptions`.
+3. `append_batches` takes as input the minted batch descriptions and written batches. Whenever the frontiers sufficiently advance,
+we take a batch description and all the batches that belong to it and append it to the persist shard.
+This also updates the shared frontier information in `storage_state` to match the persist frontier.
+This way, the controller eventually learns of the upper frontier of data written to the persist shard.
+Note that failing to `compare_and_append` must be retried, as we current expect only 1 replica to be
+running this sink operation for each source `Collection`.
+
+### Notes on the implementation
+
+- It is needlessly expensive to create a `BatchBuilder` for each timestamp.
+  - A change to this implementation is to avoid spilling data to persist for timestamps
+  whose values use a small amount of memory. These can be consolidated into large batches in `append_batches`,
+  similar to how <https://github.com/MaterializeInc/materialize/pull/17526> works.
+  - We do this in the current implementation, so doing it from many workers, instead of 1,
+  which has strictly more throughput (though does cause additional `SET` and `GET` load on AWS S3).
+  - Its undecided whether or not this should be required for the reference implementation.
+- `append_batches` and `write_batches` will be required to track statistics to expose to users.
+  - See <https://www.notion.so/materialize/User-facing-metrics-for-Sources-9603b61224a04a749969702e04662ed3> for more details
+
+
+# Rollout
+[rollout]: #rollout
+
+## Testing and observability
+[testing-and-observability]: #testing-and-observability
+
+This new implementation can be tested entirely using the existing CI, nightly, and release-qualification pipelines.
+However its important that the performance of the new implementation is carefully monitored as part of the first
+release that contains it. The standard canary-staging-production release staging will ensure we can sufficient
+experience with the new implementation before going to production broadly.
+
+## Lifecycle
+[lifecycle]: #lifecycle
+
+This implementation can be validated before release, and does not need any feature gating.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+The primary drawback of this design, as alluded to above, is that it does not represent the final implementation we want the
+storage `persist_sink` to have. Ideally, we would be able to use the exact same implementation between compute and storage, but
+a after a _large_ amount of brainstorming within the storage team, this is no feasible without significant engineering effort
+(the technical details here are mentioned in [Conclusion and alternatives](#conclusion-and-alternatives), which we are unable
+to invest at this time.
+
```

Comment:
```suggestion
An additional drawback is that the multi-worker persist sink can now write more, and smaller part files because we write those part files from each worker instead of only one worker, for each timestamp. As mentioned above, the compute `persist_sink` suffers from a similar problem but there is at least one proposed solution in <https://github.com/MaterializeInc/materialize/pull/17526>.
```

Sorry, I think you need to somehow fix up the formatting.